GDJS/GDJS/Events/CodeGeneration/BehaviorCodeGenerator.cpp

```diff
@@ -119,21 +154,51 @@ gd::String BehaviorCodeGenerator::GenerateRuntimeBehaviorTemplateCode(
     const gd::String& codeNamespace,
     std::function<gd::String()> generateInitializePropertiesCode,
     std::function<gd::String()> generatePropertiesCode,
+    std::function<gd::String()> generateInitializeSharedPropertiesCode,
+    std::function<gd::String()> generateSharedPropertiesCode,
     std::function<gd::String()> generateMethodsCode,
     std::function<gd::String()> generateUpdateFromBehaviorDataCode) {
   return gd::String(R"jscode_template(
 CODE_NAMESPACE = CODE_NAMESPACE || {};
 
+/**
+ * Shared data generated from BEHAVIOR_FULL_NAME
+ */
+CODE_NAMESPACE.RUNTIME_BEHAVIOR_CLASSNAMESharedData = class RUNTIME_BEHAVIOR_CLASSNAMESharedData {
+  constructor(sharedData) {
+    INITIALIZE_SHARED_PROPERTIES_CODE
+  }
+  
+  // Shared properties:
+  SHARED_PROPERTIES_CODE
+}
+
+CODE_NAMESPACE.getSharedData = function(instanceContainer, behaviorName) {
```

Comment:
I wonder if this should be `getRUNTIME_BEHAVIOR_CLASSNAMESharedData`?

It works now because the CODE_NAMESPACE is different for each behavior, but in theory it could be the same one day.